feat: upgrade Expo modules & react-native-reanimated v4

huhuanming opened the pull request.

huhuanming committed:
upgrade expo

huhuanming committed:
update ios pods

huhuanming committed:
react-native-worklets

huhuanming committed:
fix: fix worklets

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/15972fa9-13e1-49d6-ad5a-ea075988ac0c?refs=97dbc93546ccdfa3ede05b55a95a39ac3059c357&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/15972fa9-13e1-49d6-ad5a-ea075988ac0c/license?refs=97dbc93546ccdfa3ede05b55a95a39ac3059c357&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

huhuanming committed:
Update useFocusedTab.ts

huhuanming committed:
fix: fix version

huhuanming committed:
fix: fix lint

huhuanming committed:
fix: fix type

huhuanming committed:
Update TransactionsHistory.tsx

huhuanming committed:
Update react-native-safe-area-context to 5.6.2

Bump react-native-safe-area-context from version 5.4.1 to 5.6.2 in mobile app dependencies for improved compatibility and bug fixes.

huhuanming committed:
Merge branch 'x' into expo-modules-upgrades

Minnzen committed:
Merge branch 'x' into expo-modules-upgrades
